### DISCUSSION TITLE
Is PRBMath compatible with soilidty v0.5, and is PRBMath library used by well know projects?

### DISCUSSION THREAD

**Discussion by kevin-wad** [PaulRBerg/prb-math#221] [Q&A] [answered]
Hi, 
I have a question about your PRBMath library. 
First of all contrats it looks amazing. 

Do you know if it is used by well known projects like Uniswap or any other such project?
Do you think it would be possible to adjust it for solidity v0.5 if I only need to use the sqrt and pow functions from it. 
(I would like to be able to calculate unit256 pow (0.8)) since UD60x18 type could support 0.8 I figure it would be possible, but would it be possible using solidity v 0.5 as well? (v0.5.2 to be specific)

Thank you very much,
Kevin Wad from Etica Protocol

**Comment by PaulRBerg** [comment, marked as answer]
Hello! Thanks for your kind words and for your question.

> Do you know if it is used by well known projects like Uniswap or any other such project?

PRBMath is not used by Uniswap, but it is one of the most popular fixed-point math libraries in Solidity.

- There are currently [456 dependent projects](https://github.com/PaulRBerg/prb-math/network/dependents) on GitHub
- [>1k weekly downloads](https://www.npmjs.com/package/@prb/math) on npm
- Used by the Solidity compiler and Foundry for regression testing

> Do you think it would be possible to adjust it for solidity v0.5 if I only need to use the sqrt and pow functions from it

That would be difficult because PRBMath uses many features available only in Solidity v0.8. But it is possible to do it - you just need to slog through a refactoring process.

---

I'm curious why do you have to use v0.5?

**Reply by kevin-wad** [reply]
I'm Etica Protocol main dev. It is a fork of Ethereum Classic (for Open Source medical research without intellectual property) but dedicated to a single smart contract. The Etica smart contract was deployed with Solidity v0.5.20.
We have an upgrade process that involves hardfork to change the bytecode at Etica smart contract address.
In order to implement your library on the new smart contract I needed to update the Solidity version to v0.8.22, or if had been possible to use your library with solidity 0.5.2 I would probably have let the solidity version to 0.5.2. I wasn't sure about the consequences to change solidity version on storage layout but after tests it works. 
So no problem the smart contract will be using solidity v0.8.22

Ok thank you very much it is reassuring to know it is a well tested and used library

**Reply by PaulRBerg** [reply]
Awesome, great to hear! Any other questions, let me know.
